topic/machine-learning/llm-langchain/requirements.txt

```diff
@@ -1,17 +1,19 @@
 # Real.
 crash
 crate[sqlalchemy]
-# cratedb-toolkit
+cratedb-toolkit==0.0.3
 
-# langchain[cratedb,openai]
-# pueblo[cli,fileio,nlp]
+# langchain[cratedb,openai]==0.0.354
+# langchain-community==0.0.8
+# pueblo[cli,fileio,nlp]==0.0.4
 pydantic>=1,<3
 pypdf<4
 python-dotenv<2
 requests-cache<2
 unstructured<0.11
 
 # Development.
-cratedb-toolkit @ git+https://github.com/crate-workbench/cratedb-toolkit.git@main
+# cratedb-toolkit @ git+https://github.com/crate-workbench/cratedb-toolkit.git@main
 langchain[cratedb,openai] @ git+https://github.com/crate-workbench/langchain.git@cratedb#subdirectory=libs/langchain
+langchain-community @ git+https://github.com/crate-workbench/langchain.git@cratedb#subdirectory=libs/community
```

Comment:
That's new: After recent refactorings at LangChain, they publish separate packages for individual subsystems now, served from the same monorepo.

- https://pypi.org/project/langchain/
- https://pypi.org/project/langchain-core/
- https://pypi.org/project/langchain-community/
- https://pypi.org/project/langchain-openai/